### PR TITLE
Fixed Take event emit for ERC721 pools

### PR DESCRIPTION
**Pull request by ith-harvey**
## Description

* Noticed in takes for ERC721 pools that the collateral event emitted from take calls is inaccurate due to the integral collateral amount required for takers. This PR resolves those rounding issues by moving the `Take` event after we adjust ERC721 collateral take balances.

## Impact

* insignificant gas change

## Tasks

- [x] Changes to protocol contracts are covered by unit tests executed by CI.
- [x] Protocol contract size limits have not been exceeded.
- [x] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [x] Scope labels have been assigned as appropriate.
- [x] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

**Review by mattcushman — APPROVED** [review]
lgtm

**Review by EdNoepel — APPROVED** [review]
Much better; thank you.

File: tests/forge/unit/ERC721Pool/ERC721PoolLiquidationsTake.t.sol [hunk 1]
```diff
@@ -359,7 +359,7 @@ contract ERC721PoolLiquidationsTakeTest is ERC721HelperContract {
             maxCollateral:   2,
             bondChange:      0.243847547737474028 * 1e18,
             givenAmount:     22.183024574062103585 * 1e18,
-            collateralTaken: 1.538515492082238493 * 1e18, // not a rounded collateral, difference of 2 - 1.53 collateral should go to borrower in quote tokens at auction price
+            collateralTaken: 2.0 * 1e18, // not a rounded collateral, difference of 2 - 1.53 collateral should go to borrower in quote tokens at auction price
```
Comment (grandizzy) [inline tests/forge/unit/ERC721Pool/ERC721PoolLiquidationsTake.t.sol]: Comment still says `not a rounded collateral`

**Review by grandizzy — APPROVED** [review]
LGTM

**Review by prat-gpt — APPROVED** [review]

File: tests/forge/unit/ERC721Pool/ERC721PoolLiquidationsTake.t.sol (hunk 1 above)
Comment (prat-gpt) [reply tests/forge/unit/ERC721Pool/ERC721PoolLiquidationsTake.t.sol]: Updated [9435e57](https://github.com/ajna-finance/contracts/pull/1026/commits/9435e57178c331e693499b60ef68feccf384cd48)

**Review by dmitriia — COMMENTED** [review]
Ok, it's better to emit final qty

**Pull request merged**